plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java

```diff
@@ -239,7 +240,14 @@ private void doRegister(MLRegisterModelInput registerModelInput, ActionListener<
             if (Strings.isNotBlank(registerModelInput.getConnectorId())) {
                 connectorAccessControlHelper.validateConnectorAccess(client, registerModelInput.getConnectorId(), ActionListener.wrap(r -> {
                     if (Boolean.TRUE.equals(r)) {
-                        createModelGroup(registerModelInput, listener);
+                        if (registerModelInput.getModelInterface() == null) {
+                            mlModelManager.getConnector(registerModelInput.getConnectorId(), ActionListener.wrap(connector -> {
```

Comment:
can't you stash thread context here before invoking this method?

Reply:
Well yeah this is something we can try